vitrina/helpers.py

```diff
@@ -449,41 +458,137 @@ def get_current_domain(request: WSGIRequest, ensure_secure=False) -> str:
+    list_keys = base_template_content[base_template_content.find("{") + 1:base_template_content.rfind("}")].replace(
+        '{', '').replace('}', '').split()
+    email_template_to_save = base_template_content
+    if email_template_keys:
+        for key in list_keys:
+            if key in email_template_keys.keys():
+                if email_template_keys[key] is not None:
+                    base_template_content = base_template_content.replace("{" + key + "}", email_template_keys[key])
+            else:
+                base_template_content = base_template_content.replace("{" + key + "}", '')
     if not email_template:
         email_subject = email_title = email_title_subject
-        email_content = base_template_content.format(*email_template_keys)
+        email_content = base_template_content
         created_template = EmailTemplate.objects.create(
             created=datetime.datetime.now(),
             version=0,
             identifier=email_identifier,
-            template=base_template_content,
+            template=email_template_to_save,
             subject=_(email_title_subject),
             title=_(email_title)
         )
         created_template.save()
     else:
         email_template = email_template.first()
         email_content = str(email_template.template)
-        email_content = email_content.format(*email_template_keys)
-        email_subject = str(email_template.subject)
+        if email_template_keys:
+            for key in list_keys:
+                if key in email_template_keys.keys():
+                    if email_template_keys[key] is not None:
+                        email_content = email_content.replace("{" + key + "}", email_template_keys[key])
+                else:
+                    email_content = email_content.replace("{" + key + "}", '')
+            email_subject = str(email_template.subject)
+
 
-    return {'email_content': email_content, 'email_subject': email_subject}
+def _get_email_tempate_from_db(name: str) -> EmailTemplate | None:
+    try:
+        return EmailTemplate.objects.get(identifier=name)
+    except EmailTemplate.DoesNotExist:
+        return None
+
+
+def email(
+    recipients: list[str],
+    email_identifier: str,
+    name: str,  # template name
+    context: dict[str, Any] | None = None,
+    *,
+    # Allow user to override email templates via Admin.
+    override: bool = True,
+) -> dict[str, str]:
+    context = context or {}
+    data_dict = {}
+    email_template = None
+    email_send = False
+    subject = None
+    if override:
+        email_template = _get_email_tempate_from_db(email_identifier)
+    if email_template:
+        from unidecode import unidecode
+        subject = email_template.subject
+        template_db = email_template.template
+        template_db = unidecode(template_db)
+        template = Template(template_db)
+        for key in template.nodelist:
+            if type(key).__name__ == 'VariableNode':
```

Comment:
```suggestion
            if isinstance(key, VariableNode):
```

See [isinstance](https://docs.python.org/3/library/functions.html#isinstance) and [Why use isinstance instead of type?](https://mathspp.com/blog/twitter-threads/why-use-isinstance-instead-of-type).

Reply:
The idea of this check is not exactly to check variable type, cause 'VariableNode' is not exactly the type itself but rather type name, so in this case it seems that isinstance is not working for this case.

Reply:
[VariableNode](https://github.com/django/django/blob/d9b91e38361696014bdc98434d6d018eae809519/django/template/base.py#L1056) is exactly a type and `key` might be instance of `VariableNode` type.

I don't understand, purpose of this block. If I understand it correctly, you can simply do

```python
context = Context(context)
```